### PR TITLE
REF add soname to libgfortran package name

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -86,10 +86,10 @@ outputs:
...
       host:
         - sysroot_{{ ctng_target_platform }} {{ conda_glibc_ver }}
-        - {{ pin_subpackage("libgomp", exact=True) }}           # [target_platform == ctng_target_platform]
-        - {{ pin_subpackage("libstdcxx-ng", exact=True) }}      # [target_platform == ctng_target_platform]
-        - {{ pin_subpackage("libgcc-ng", exact=True) }}         # [target_platform == ctng_target_platform]
-        - {{ pin_subpackage("libgfortran-ng") }}                # [target_platform == ctng_target_platform]
+        - {{ pin_subpackage("libgomp", exact=True) }}                       # [target_platform == ctng_target_platform]
+        - {{ pin_subpackage("libstdcxx-ng", exact=True) }}                  # [target_platform == ctng_target_platform]
+        - {{ pin_subpackage("libgcc-ng", exact=True) }}                     # [target_platform == ctng_target_platform]
+        - {{ pin_subpackage("libgfortran" ~ libgfortran_soname ~ "-ng") }}  # [target_platform == ctng_target_platform]
```
Comment (beckermr) [inline recipe/meta.yaml:92]:
```suggestion
        - {{ pin_subpackage("libgfortran" ~ libgfortran_soname ~ "-ng", exact=True) }}  # [target_platform == ctng_target_platform]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -195,16 +195,16 @@ outputs:
       run:
         # For cpp and crt{i,n}.o
         - gcc_impl_{{ ctng_target_platform }} >={{ ctng_gcc }}
-        - {{ pin_subpackage("libgfortran-ng") }}                # [target_platform == ctng_target_platform]
-        - {{ pin_subpackage("libgcc-ng", max_pin=None) }}       # [target_platform == ctng_target_platform]
+        - {{ pin_subpackage("libgfortran" ~ libgfortran_soname ~ "-ng") }}  # [target_platform == ctng_target_platform]
```
Comment (beckermr) [inline recipe/meta.yaml:198]:
```suggestion
        - libgfortran{{ libgfortran_soname }}-ng {{ ctng_gcc }} *_{{ build_num | default(0) }} # [target_platform == ctng_target_platform]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -86,10 +86,10 @@ outputs:
...
       host:
         - sysroot_{{ ctng_target_platform }} {{ conda_glibc_ver }}
-        - {{ pin_subpackage("libgomp", exact=True) }}           # [target_platform == ctng_target_platform]
-        - {{ pin_subpackage("libstdcxx-ng", exact=True) }}      # [target_platform == ctng_target_platform]
-        - {{ pin_subpackage("libgcc-ng", exact=True) }}         # [target_platform == ctng_target_platform]
-        - {{ pin_subpackage("libgfortran-ng") }}                # [target_platform == ctng_target_platform]
+        - {{ pin_subpackage("libgomp", exact=True) }}                       # [target_platform == ctng_target_platform]
+        - {{ pin_subpackage("libstdcxx-ng", exact=True) }}                  # [target_platform == ctng_target_platform]
+        - {{ pin_subpackage("libgcc-ng", exact=True) }}                     # [target_platform == ctng_target_platform]
+        - {{ pin_subpackage("libgfortran" ~ libgfortran_soname ~ "-ng", exact=True) }}  # [target_platform == ctng_target_platform]
```
Comment (beckermr) [inline recipe/meta.yaml:92]:
```suggestion
        - libgfortran{{ libgfortran_soname }}-ng {{ ctng_gcc }} *_{{ build_num | default(0) }}  # [target_platform == ctng_target_platform]
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -372,6 +372,14 @@ outputs:
...
 
   - name: libgfortran-ng
+    target: {{ ctng_target_platform }}
+    build:
+      skip: True   # [target_platform != ctng_target_platform]
+    requirements:
+      run:
+        - libgfortran{{ libgfortran_soname }}-ng {{ ctng_gcc }}.*
+
+  - name: libgfortran{{ libgfortran_soname }}-ng
```
Comment (isuruf) [inline recipe/meta.yaml:382]: Can we remove `-ng` here?

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@xhochy @isuruf I put some license fixes in this PR to try and make things SPDX. Can you all look over them?

**Comment by xhochy** [comment]
> @xhochy @isuruf I put some license fixes in this PR to try and make things SPDX. Can you all look over them?

They look matching 👍

**Review by isuruf — APPROVED** [review]

**Pull request merged**